### DISCUSSION TITLE
(pdf/lua)latex

### DISCUSSION THREAD

**Discussion by stenw** [texstudio-org/texstudio#2939] [Q&A]
Using texstudio 4.5.1 under windows 11, when I compile the document using any of the commands under Tools/Commands (latex/pdflatex/luatex/xetex) it seems that alway the default compilation method is chosen instead of the method indicated. Am I mising a setting?

**Comment by stenw** [comment]
Important additional information. This does not happen in all documents so it might not be related to the texstudio application. 
In a document where this does happen the compilation method can still be changed ny changing the default method.

**Comment by sunderme** [comment]
works here as expected. What exactly do you do ?

**Reply by stenw** [reply]
I just go to the menu and select either latex, pdflatex or luatex and see that the command that is used is a different one (whatever was selected as the default). I have copied the whole text in a new document and there it does not give me any problem. I expect that this will be hard to replicate. If there is nobody else with this problem maybe ignore this issue for now.

**Comment by octaeder** [comment]
I've newer seen such a problem and I can't reproduce with win10. I often switch between lua and pdf LaTeX compiler. Sometimes I do so by changing the default compiler in the config, sometimes I use the item needed from the tools menu. And I have added pdflatex from the Tools menu to the Tools toolbar (s. config) for quicker access. I compile files as well as code entered in new editors.

Do you have some magic comment in the document? As an example: % !TeX program = pdflatex
